app/actions/websocket/users.ts

```diff
@@ -0,0 +1,102 @@
+                const displaySettings = getTeammateNameDisplaySetting(preferences, config, license);
+                const models: Model[] = [];
+                channels.forEach(async (channel) => {
+                    const dbProfiles = await database.database.get<UserModel>(USER).query(Q.on(CHANNEL_MEMBERSHIP, Q.where('channel_id', channel.id))).fetch();
+                    const newProfiles: Array<UserModel|UserProfile> = dbProfiles.filter((u) => u.id === user.id);
+                    newProfiles.push(user);
+                    const newDisplayName = displayGroupMessageName(newProfiles, currentUser.locale, displaySettings, currentUser.id);
+                    if (channel.displayName !== newDisplayName) {
+                        channel.prepareUpdate((c) => {
+                            c.displayName = newDisplayName;
+                        });
+                    }
+                    models.push(channel);
+                });
+                if (models.length) {
+                    database.operator.batchRecords(models);
+                }
+            }
+        }
+    } else {
+        database.operator.handleUsers({users: [user], prepareRecordsOnly: false});
+
+        const channels = await database.database.get<ChannelModel>(CHANNEL).query(
+            Q.where('type', Q.oneOf([General.DM_CHANNEL, General.GM_CHANNEL])),
+            Q.on(CHANNEL_MEMBERSHIP, Q.where('user_id', user.id))).fetch();
+        if (!channels?.length) {
+            return;
+        }
+
+        const {config, license} = await queryCommonSystemValues(database.database);
+        const preferences = await queryPreferencesByCategoryAndName(database.database, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.NAME_NAME_FORMAT);
+        const displaySettings = getTeammateNameDisplaySetting(preferences, config, license);
+        const models: Model[] = [];
+        channels?.forEach(async (channel) => {
+            let newDisplayName = '';
+            if (channel.type === General.DM_CHANNEL) {
+                newDisplayName = displayUsername(user, currentUser.locale, displaySettings);
+            } else {
+                const dbProfiles = await database.database.get<UserModel>(USER).query(Q.on(CHANNEL_MEMBERSHIP, Q.where('channel_id', channel.id))).fetch();
+                const newProfiles: Array<UserModel|UserProfile> = dbProfiles.filter((u) => u.id === user.id);
+                newProfiles.push(user);
+                newDisplayName = displayGroupMessageName(newProfiles, currentUser.locale, displaySettings, currentUser.id);
+            }
+
+            if (channel.displayName !== newDisplayName) {
+                channel.prepareUpdate((c) => {
+                    c.displayName = newDisplayName;
+                });
+                models.push(channel);
+            }
+        });
```

Comment:
I wonder if this can be a local action and then call it also from L#42. The function can have an argument to determine if it should do it for GM's & DM's